js/kkex.js

```diff
@@ -0,0 +1,408 @@
+        let symbol = undefined;
+        if (typeof market !== 'undefined')
+            symbol = market['symbol'];
+        let side = this.safeString (order, 'side');
+        if (typeof side === 'undefined') {
+            side = this.safeString (order, 'type');
+        }
+        let timestamp = undefined;
+        let iso8601 = undefined;
+        if ('datetime' in order) {
+            timestamp = this.parse8601 (order['datetime']);
+            iso8601 = this.iso8601 (timestamp);
+        }
+        return {
+            'id': order['order_id'] || order['id'],
+            'datetime': iso8601,
+            'timestamp': timestamp,
+            'lastTradeTimestamp': undefined,
+            'status': 'open',
+            'symbol': symbol,
+            'type': 'limit',
+            'side': side,
+            'price': order['price'],
+            'cost': undefined,
+            'amount': order['number'],
+            'filled': undefined,
+            'remaining': undefined,
+            'trades': undefined,
+            'fee': undefined,
+            'info': order,
+        };
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        await this.loadMarkets ();
+        let market = this.market (symbol);
+        let sides = ['buy', 'sell', 'buy_market', 'sell_market'];
+        let request = { 'symbol': market['id'] };
+        if (type === 'market') {
+            if (side === 'sell') {
+                request['amount'] = amount;
+            } else if (side === 'buy') {
+                request['price'] = amount;
+            }
+            side += '_market';
+        } else {
+            request['amount'] = amount;
+            request['price'] = price;
+        }
+        request['type'] = side;
+        if (sides.indexOf (side) === -1) {
+            throw new ExchangeError ('side not in', sides);
+        }
+        let response = await this.privatePostTrade (this.extend (request, params));
+        if (response['result'] === false) {
+            throw new ExchangeError (JSON.stringify (response));
+        }
+        let id = response['order_id'];
+        let order = this.parseOrder ({
+            'id': id,
+            'price': price,
+            'number': amount,
+            'side': side,
+        }, market);
+        this.orders[id] = order;
+        return order;
+    }
+
+    async cancelOrder (id, symbol = undefined, params = {}) {
+        await this.loadMarkets ();
+        let response = await this.privatePostCancelOrder (this.extend ({
+            'order_id': id,
+            'symbol': symbol.replace ('/', ''),
+        }, params));
+        return response;
+    }
+
+    async fetchOpenOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        return await this.fetchOrders (0, symbol, since, limit, params);
+    }
+
+    async fetchOrders (status = undefined, symbol = undefined, since = undefined, limit = undefined, params = {}) {
```

Comment:
First of, thanks for your work! Much appreciated!

Unfortunately, we can't change signatures of unified methods (the ordering of arguments and the set of arguments in itself) – they should be derived from the base class, otherwise it won't transpile properly to python and php. Can you change the signature of it, plz, so to make the Travis test pass? Thx!

Reply:
i got other error 

```
./python/ccxt/kkex.py:114:33: E251 unexpected spaces around keyword / parameter equals
./python/ccxt/kkex.py:114:35: E225 missing whitespace around operator
./python/ccxt/kkex.py:114:35: E999 SyntaxError: invalid syntax
./python/ccxt/kkex.py:189:29: E251 unexpected spaces around keyword / parameter equals
./python/ccxt/kkex.py:189:31: E225 missing whitespace around operator
./python/ccxt/kkex.py:300:31: E712 comparison to False should be 'if cond is False:' or 'if not cond:'
./python/ccxt/kkex.py:352:39: E251 unexpected spaces around keyword / parameter equals
./python/ccxt/kkex.py:352:41: E225 missing whitespace around operator
./python/ccxt/kkex.py:353:18: E251 unexpected spaces around keyword / parameter equals
./python/ccxt/kkex.py:353:20: E251 unexpected spaces around keyword / parameter equals
./python/ccxt/kkex.py:354:22: E251 unexpected spaces around keyword / parameter equals
./python/ccxt/kkex.py:354:24: E251 unexpected spaces around keyword / parameter equals
./python/ccxt/async/kkex.py:114:33: E251 unexpected spaces around keyword / parameter equals
./python/ccxt/async/kkex.py:114:35: E225 missing whitespace around operator
./python/ccxt/async/kkex.py:114:35: E999 SyntaxError: invalid syntax
./python/ccxt/async/kkex.py:189:29: E251 unexpected spaces around keyword / parameter equals
./python/ccxt/async/kkex.py:189:31: E225 missing whitespace around operator
./python/ccxt/async/kkex.py:300:31: E712 comparison to False should be 'if cond is False:' or 'if not cond:'
./python/ccxt/async/kkex.py:352:39: E251 unexpected spaces around keyword / parameter equals
./python/ccxt/async/kkex.py:352:41: E225 missing whitespace around operator
./python/ccxt/async/kkex.py:353:18: E251 unexpected spaces around keyword / parameter equals
./python/ccxt/async/kkex.py:353:20: E251 unexpected spaces around keyword / parameter equals
./python/ccxt/async/kkex.py:354:22: E251 unexpected spaces around keyword / parameter equals
./python/ccxt/async/kkex.py:354:24: E251 unexpected spaces around keyword / parameter equals
ERROR: InvocationError for command '/Users/jk/bixin/ccxt/python/.tox/qa/bin/flake8' (exited with code 1)
___________________________________________________________________ summary ___________________________________________________________________
ERROR:   qa: commands failed
```

Reply:
should i fix it ?

Reply:
@ericjjj yes, if you can – it would be really very helpful of you!
Basically, we should be very careful with spacing and indentation (tabs), as Python is very critical to this. To make sure that everything is correct, you can use the following rules:

https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#derived-exchange-classes

Let me know, if you have any questions! Thx!

Reply:
thank you, I will fix it

Reply:
@ericjjj great! Thx so much!